sql/core/src/test/scala/org/apache/spark/sql/execution/SparkPlanSuite.scala

```diff
@@ -109,4 +113,19 @@ class SparkPlanSuite extends QueryTest with SharedSparkSession {
     assert(err.getMessage.contains("Deduplicate operator for non streaming data source " +
       "should have been replaced by aggregate in the optimizer"))
   }
+
+  test("SPARK-37221: The collect-like API in SparkPlan should support columnar output") {
+    val results = ColumnarOp(LocalTableScanExec(Nil, Nil)).executeCollect()
+    assert(results.isEmpty)
```

Comment:
I think we are pretty sure this PR change is correct, but shall we add another test case to double check the non-empty result?

Reply:
Sure. Let me add another one.